### ISSUE TITLE
Python 3.13.10 regression when validating SAM templates

### ISSUE THREAD

**Issue by btyrrell-mn** [aws-cloudformation/cfn-lint#4310]
### CloudFormation Lint Version

1.42.0

### What operating system are you using?

Reported on Mac and Ubuntu

### Describe the bug

I came to report a regression with Python 3.13.10 as well. I had a bug report open with astral-sh at first.

https://github.com/astral-sh/python-build-standalone/issues/905#issuecomment-3614348606

`cfn-lint` incorrectly reports errors for valid templates.

### Expected behavior

A valid template should pass. `cfn-lint` running on Python 3.13.10 returns an `E0001` error for the first line. It claims event properties for API functions are missing.

```
E0001 Error transforming template: Resource with id [ApiFunctionEvent] is invalid. Missing required property 'Path'.
template.yaml:1:1
```

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:
  ApiFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: python3.13
      InlineCode: |
        def handler(event, context):
            return {
                'statusCode': 200,
                'body': 'Hello World'
            }
      Events:
        RootPath:
          Type: Api
          Properties:
            Method: GET
            Path: /
```
